Issues compiling elektra on OpenBSD 5.8

* OpenBSD doesn't use/know EBADMSG
```
/home/thomas/libelektra/src/plugins/resolver/resolver.c: In function 'elektraAddErrnoText':
/home/thomas/libelektra/src/plugins/resolver/resolver.c:278:20: error: 'EBADMSG' undeclared (first use in this function)
  else if (errno == EBADMSG)
                    ^
/home/thomas/libelektra/src/plugins/resolver/resolver.c:577:12: error: 'EBADMSG' undeclared (first use in this function)
    errno = EBADMSG;
            ^
```

* OpenBSD doesn't have librt
should be 
```
if (UNIX AND NOT APPLE AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "OpenBSD")
	set(CMAKE_REALTIME_LIBS_INIT rt)
endif()
```
in cmake/ElektraCompiling.cmake

Found another issue on OpenBSD (tried to compile a67c05505552c57861125c3ed4bf26cda7553fdc )

	[  1%] Building CXX object src/error/CMakeFiles/elektra-export-errors.dir/exporterrors.cpp.o
	cc1plus: warning: -Wuninitialized is not supported without -O
	In file included from /home/pnirschl/elektra/src/error/exporterrors.cpp:9:
	/home/pnirschl/elektra/src/error/parser.hpp:29: error: expected ';' before 'override'
	/home/pnirschl/elektra/src/error/parser.hpp:33: error: expected `;' before '}' token
	/home/pnirschl/elektra/src/error/parser.hpp: In constructor 'parse_error::parse_error(std::string, int)':
	/home/pnirschl/elektra/src/error/parser.hpp:22: error: 'move' is not a member of 'std'
	/home/pnirschl/elektra/src/error/exporterrors.cpp: In function 'int main(int, char**)':
	/home/pnirschl/elektra/src/error/exporterrors.cpp:281: error: 'to_string' was not declared in this scope
	/home/pnirschl/elektra/src/error/exporterrors.cpp:284: error: no matching function for call to 'std::basic_ofstream<char, std::char_traits<char> >::basic_ofstream(std::string&)'
	/usr/include/g++/fstream:594: note: candidates are: std::basic_ofstream<_CharT, _Traits>::basic_ofstream(const char*, std::_Ios_Openmode) [with _CharT = char, _Traits = std::char_traits<char>]
	/usr/include/g++/fstream:578: note:                 std::basic_ofstream<_CharT, _Traits>::basic_ofstream() [with _CharT = char, _Traits = std::char_traits<char>]
	/usr/include/g++/iosfwd:92: note:                 std::basic_ofstream<char, std::char_traits<char> >::basic_ofstream(const std::basic_ofstream<char, std::char_traits<char> >&)

--- a/cmake/ElektraCompiling.cmake
+++ b/cmake/ElektraCompiling.cmake
@@ -121,7 +121,9 @@ if (ENABLE_COVERAGE)
 	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fprofile-arcs -ftest-coverage -lgcov")
 endif (ENABLE_COVERAGE)
 
-if (UNIX AND NOT APPLE)
+# check if native librt is available
+find_library(CMAKE_HAS_LIBRT "rt")
+if (CMAKE_HAS_LIBRT)
 	set(CMAKE_REALTIME_LIBS_INIT rt)
 endif()
 

--- a/src/plugins/resolver/resolver.c
+++ b/src/plugins/resolver/resolver.c
@@ -272,7 +272,7 @@ static void elektraAddErrnoText (char * errorText)
 	{
 		strcpy (buffer, "could not find a / in the pathname");
 	}
-	else if (errno == EBADMSG)
+	else if (errno == EINVAL)
 	{
 		strcpy (buffer, "went up to root for creating directory");
 	}
@@ -568,7 +568,7 @@ static int elektraMkdirParents (resolverHandle * pk, const char * pathname, Key 
 		{
 			// set any errno, corrected in
 			// elektraAddErrnoText
-			errno = EBADMSG;
+			errno = EINVAL;
 			goto error;
 		}
 

